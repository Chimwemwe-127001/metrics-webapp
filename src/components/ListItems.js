import { useSelector } from 'react-redux';
import { CircularProgress } from 'react-cssfx-loading';
import { Link } from 'react-router-dom';
import { Input } from 'semantic-ui-react';
import { useState } from 'react';

const ListItems = () => {
  const countries = useSelector((state) => state.covidReducer.oneCountry);
  const [filteredResults, setFilteredResults] = useState([]);
  const [searchInput, setSearchInput] = useState('');
  const searchItems = (searchValue) => {
    setSearchInput(searchValue);
    if (searchInput !== '') {
      const filteredData = countries.filter((item) => Object.values(item).join('').toLowerCase().includes(searchInput.toLowerCase()));
      setFilteredResults(filteredData);
    } else {
      setFilteredResults(countries);
    }
  };
  const populate = countries
    ? countries.map((country) => (
      <div key={country.id} className="country d-flex">
        <Link
          className="navLink"
          to={{
            pathname: `/itemsdetails/${country.name}`,
          }}
        >
          <h4 className="companyName">{country.name}</h4>
        </Link>
        <div>
          <p className="casesConfirmedVal">{country.confirmedCases}</p>
          <p className="casesConfirmed">cases confirmed</p>
        </div>
      </div>
    ))
    : <div className="spinLoad"><CircularProgress className="BarWave" color="#fff" width="20%" duration="2s" /></div>;
  return (
    <div className="list-container">
      <h2>Search by Country name</h2>
      <Input
        className="search"
        icon="search"
        placeholder="Search..."
        onChange={(e) => searchItems(e.target.value)}
      />
      <div className="list-countries d-flex">
        {searchInput.length > 1
          ? (
            filteredResults.map((country) => (
              <div key={country.id} className="country d-flex">
                <Link
                  className="navLink"
                  to={{
                    pathname: `/itemsdetails/${country.name}`,
                  }}
                >
                  <h4 className="companyName">{country.name}</h4>
                </Link>
                <div>
                  <p className="casesConfirmedVal">{country.confirmedCases}</p>
                  <p className="casesConfirmed">cases confirmed</p>
                </div>
              </div>
            ))
          )
          : populate}
      </div>
    </div>

  );
};

export default ListItems;
