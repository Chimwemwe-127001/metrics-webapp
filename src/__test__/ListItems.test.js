import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import ListItems from '../components/ListItems';

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('testing the app', () => {
  it('testing ListItems UI', () => {
    act(() => {
      render(
        <Provider store={store}>
          <ListItems />
        </Provider>, container,
      );
    });

    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });
});
