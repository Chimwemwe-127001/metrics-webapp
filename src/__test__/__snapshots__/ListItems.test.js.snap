// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the app testing ListItems UI 1`] = `
"<div class=\\"list-container\\">
  <h2>Search by Country name</h2>
  <div class=\\"ui icon input search\\"><input placeholder=\\"Search...\\" type=\\"text\\"><i aria-hidden=\\"true\\" class=\\"search icon\\"></i></div>
  <div class=\\"list-countries d-flex\\">
    <div class=\\"spinLoad\\"><svg crossorigin=\\"anonymous\\" viewBox=\\"25 25 50 50\\" style=\\"--width: 20%; --height: 3em; --color: #fff; --duration: 2s;\\" class=\\"cssfx-circular-progress-svg BarWave\\">
        <circle class=\\"cssfx-circular-progress-circle\\" cx=\\"50\\" cy=\\"50\\" r=\\"20\\"></circle>
      </svg></div>
  </div>
</div>"
`;
